ground_truth_labeling_jobs/ground_truth_conversion_scripts/GT_semantic_segmentation_to_COCO.ipynb

```diff
@@ -0,0 +1,443 @@
+    "    colors = np.unique(a.reshape(-1, a.shape[2]), axis=0)\n",
+    "    segments = []\n",
+    "    for i in range(colors.shape[0]):\n",
+    "        if colors[i][:-1].mean() < 1:\n",
+    "            segments.append(colors[i])\n",
+    "    \n",
+    "    for idds, seg in enumerate(range(len(segments))):\n",
+    "\n",
+    "        color_hex = matplotlib.colors.to_hex(segments[seg])\n",
+    "        category_id = mapping[color_hex]\n",
+    "        img = io.imread(img_file, plugin='matplotlib')\n",
+    "        msk = segments[seg]  \n",
+    "        masked_img = cv2.inRange(img, msk, msk)\n",
+    "        \n",
+    "        if len(segments) > 1:\n",
+    "            img[masked_img>0]=(255,255,255)\n",
+    "        annot = generate_segmentation(img,category_id, idd, image_id)\n",
+    "        idd+=1\n",
+    "        annotations.append(annot)\n",
+    "print(annotations)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Construct the input COCO file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "COCO_json = {\n",
+    "    \"licenses\": licenses,\n",
+    "    \"info\": info,\n",
+    "    \"categories\": categories,\n",
+    "    \"images\": images,\n",
+    "    \"annotations\": annotations\n",
+    "    \n",
+    "}\n",
+    "with open('COCO_file.json', 'w') as json_file:\n",
+    "        json.dump(COCO_json, json_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test and Visualize Each Unique Category Segmentation\n",
+    "\n",
+    "This section is based on this notebook https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocoDemo.ipynb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "from pycocotools.coco import COCO\n",
+    "import numpy as np\n",
+    "import skimage.io as io\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pylab\n",
+    "pylab.rcParams['figure.figsize'] = (20.0, 10.0)\n",
+    "\n",
+    "annFile='COCO_file.json'\n",
+    "coco=COCO(annFile)\n",
+    "# display COCO categories and supercategories\n",
+    "cats = coco.loadCats(coco.getCatIds())\n",
+    "nms=[cat['name'] for cat in cats]\n",
+    "print('COCO categories: \\n{}\\n'.format(' '.join(nms)))\n",
+    "\n",
+    "nms = set([cat['supercategory'] for cat in cats])\n",
+    "print('COCO supercategories: \\n{}'.format(' '.join(nms)))\n",
+    "\n",
+    "\n",
+    "catIds = coco.getCatIds(catNms=['bike','motorbike']);\n",
+    "imgIds = coco.getImgIds(catIds=catIds );\n",
+    "imgIds = coco.getImgIds(imgIds = [1]) #<<<<<<<<<<<< Change the number for different test images\n",
+    "img = coco.loadImgs(imgIds[np.random.randint(0,len(imgIds))])[0]\n",
+    "print(img)\n",
+    "\n",
+    "I = io.imread(img['file_name'])\n",
```

Comment:
This doesn't work for me, unless I download the original (unannotated) images into the working directory. Maybe either add the logic to due this by retrieving the input manifest, or at least give a warning to the user?

Reply:
Fixed